### DISCUSSION TITLE
Merge Editor Issues

### DISCUSSION THREAD

**Discussion by inchara-1998** [microsoft/vscode-discussions#315] [Extension Development QnA] [answered]
Hi I am building new vs code extension for our source control project we need merge editor for merging functionality.
I am using "vscode.commands.executeCommand('_open.mergeEditor', options);" for opening merge editor.
i came out through when scenario while doing Accept Merge operation
If user try to click Accept merge button if still there are conflicts unhandled. i am trying to close the merge editor before merge at that time it will trigger pop up box like there are still conflicts.
image
I am using this code to close editor
let didCloseTab = await vscode.window.tabGroups.close(mergeEditorTabs, true);
if its true it will continue merge

But if user select close with conflicts its closing merge editor so its setting "didCloseTab" value true and accepting merge.
![image](https://user-images.githubusercontent.com/118963451/207391262-e14f82d8-6a4d-466e-906d-71a90ea2ffec.png)

I want to identify the option which user is clicking because i need to restrict merge option in this case.
How can i do that??
is there any event i can use to capture that.

OR

Is there any way apart from using this command for "vscode.commands.executeCommand('_open.mergeEditor', options);"

to open merge editor and use it from VS code.?

Can anyone please suggest any alternative way to implement merge.

**Comment by isidorn** [comment, marked as answer]
We have not really opened this up through our API. And I am not sure if there is another command to achieve opening the mergeEditor.

@hediet might have more details.

**Comment by inchara-1998** [comment]
How can i detect remaining conflicts shown in merge editor on right section of result file in my extension code.

Is there any property i can access??

![image](https://user-images.githubusercontent.com/118963451/207788791-abf957e5-ea30-4599-b653-1a1cb7d74066.png)

**Reply by isidorn** [reply]
I do not think there is.

**Comment by inchara-1998** [comment]
Detecting conflict is one of the important use case in our extension's merge functionality. Is there any proposed APIs which we can use or any work around to detect remaining conflicts in Merge Editor?

**Reply by isidorn** [reply]
This is not exposed as API right now. I am not aware of any work arounds.
You can spawn git and try to communicate with it directly for example.

**Comment by davidchase** [comment]
Hi @isidorn is this 3 way merge still not exposed via the API?

**Reply by isidorn** [reply]
Correct.

**Reply by davidchase** [reply]
are there plans? 

Seems like a useful feature and it looks like we have to either use this internal command approach or possibly use a child_process and call `code merge`

**Reply by isidorn** [reply]
No plans in the next 12 months.
